### PR TITLE
Add Qwen-Image-Edit Inpainting pipeline

### PR DESCRIPTION
**Pull request by Trgtuan10**
# What does this PR do?

This PR introduces support for the **Qwen-Image-Edit** model in **Inpainting** tasks, expanding the model’s creative capabilities and integration within the Diffusers library.

## Example code
```python
import torch
from diffusers import QwenImageEditInpaintPipeline
from diffusers.utils import load_image
import os
os.environ["HF_ENABLE_PARALLEL_LOADING"] = "YES"

pipe = QwenImageEditInpaintPipeline.from_pretrained("Qwen/Qwen-Image-Edit", torch_dtype=torch.bfloat16)
pipe.to("cuda")
prompt = "change the hat to red"
negative_prompt = " "
source = load_image("https://github.com/Trgtuan10/Image_storage/blob/main/cute_cat.png?raw=true")
mask = load_image("https://github.com/Trgtuan10/Image_storage/blob/main/mask_cat.png?raw=true")

image = pipe(
    prompt=prompt,
    negative_prompt=negative_prompt,
    image=source,
    mask_image=mask,
    strength=1.0,
    num_inference_steps=35,
    true_cfg_scale=4.0,
    generator=torch.Generator(device="cuda").manual_seed(422)
).images[0]
image.save(f"qwen_inpainting.png")
```
## Comparison about performance of QwenImage-Edit Inpaint

<table>
  <tr>
    <td width="50%"><b>Init image</b><br/>
      
    </td>
    <td width="50%"><b>Mask</b><br/>
      
    </td>
  </tr>
</table>

<table>
  <tr>
    <td><b>QwenImage Inpaint</b><br/></td>
    <td><b>QwenImage-Edit</b><br/></td>
    <td><b>QwenImage-Edit Inpaint</b><br/></td>
  </tr>
</table>

## Who can review?

- Pipelines and pipeline callbacks: @yiyixuxu and @asomoza

cc @a-r-r-o-w @sayakpaul

### CODE REVIEW

**Comment by sayakpaul** [comment]
Cc: @naykun as well.

**Comment by naykun** [comment]
Hi @Trgtuan10 , that would be an amazing feature! It seems to preserve details better than the current `QwenImageInpaintPipeline`—great improvement!  

I’ll dive in and run some more extensive tests on this PR shortly. Thanks so much for your contribution, and for helping push this forward! 🙌

**Comment by TuanNT-ZenAI** [comment]
Thanks @naykun! Please feel free to test it out and let me know if you run into any issues.

**Comment by TuanNT-ZenAI** [comment]
Hi @naykun @sayakpaul, have you tested it yet? I'd appreciate any feedback or reviews!

**Comment by naykun** [comment]
@Trgtuan10 I've tested it, and things are working well overall. This approach shows promise for handling complex editing scenarios while keeping the rest of the content unchanged. There are a few unstable cases where instructions don't always take effect, but I believe we can mitigate those with better prompts and strategic masking. I suggest we move forward with this PR.
cc @sayakpaul @yiyixuxu

File: src/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_inpaint.py [hunk 1]
```diff
@@ -0,0 +1,1063 @@
...
+        attention_kwargs: Optional[Dict[str, Any]] = None,
+        callback_on_step_end: Optional[Callable[[int, int, Dict], None]] = None,
+        callback_on_step_end_tensor_inputs: List[str] = ["latents"],
+        max_sequence_length: int = 512,
+    ):
+        r"""
+        Function invoked when calling the pipeline for generation.
+
+        Args:
+            prompt (`str` or `List[str]`, *optional*):
```
Comment (sayakpaul) [inline src/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_inpaint.py:696]: Seems like `image` entry is missing here?

File: src/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_inpaint.py [hunk 2]
```diff
@@ -0,0 +1,1063 @@
...
+        mask_image: PipelineImageInput = None,
+        masked_image_latents: PipelineImageInput = None,
+        true_cfg_scale: float = 4.0,
+        height: Optional[int] = None,
+        width: Optional[int] = None,
+        padding_mask_crop: Optional[int] = None,
+        strength: float = 0.6,
+        num_inference_steps: int = 50,
+        sigmas: Optional[List[float]] = None,
+        guidance_scale: float = 1.0,
```
Comment (sayakpaul) [inline src/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_inpaint.py]: For `guidance_scale` we're defaulting to `None` for now. Check: https://github.com/huggingface/diffusers/pull/12223

**Review by sayakpaul — COMMENTED** [review]
Thanks for this! I have two minor comments. 

I think we should also add tests and docs. Thanks to @naykun for running the tests as well.

File: src/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_inpaint.py [hunk 3]
```diff
@@ -0,0 +1,1063 @@
...
+        width = 2 * (int(width) // (vae_scale_factor * 2))
+
+        latents = latents.view(batch_size, height // 2, width // 2, channels // 4, 2, 2)
+        latents = latents.permute(0, 3, 1, 4, 2, 5)
+
+        latents = latents.reshape(batch_size, channels // (2 * 2), 1, height, width)
+
+        return latents
+
+    def _encode_vae_image(self, image: torch.Tensor, generator: torch.Generator):
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_inpaint.py:429]: is this one not `#Copied from` ?

File: src/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_inpaint.py [hunk 4]
```diff
@@ -0,0 +1,1063 @@
...
+        self.vae.enable_tiling()
+
+    def disable_vae_tiling(self):
+        r"""
+        Disable tiled VAE decoding. If `enable_vae_tiling` was previously enabled, this method will go back to
+        computing decoding in one step.
+        """
+        self.vae.disable_tiling()
+
+    def prepare_latents(
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_inpaint.py:494]: is this different from the QwenImage Inpaint? if not, let's `#Copied from`

**Review by yiyixuxu — APPROVED** [review]
thank you for your PR!! 
left some very small comments about `#Copied from`

**Comment by Trgtuan10** [comment]
Thanks for the reviews @sayakpaul  @yiyixuxu. I’ve updated the code to stay up to date with the latest changes from the main branch. Please check it again.

**Comment by yiyixuxu** [comment]
@bot /style

**Comment by yiyixuxu** [comment]
oh CI still fail and style bot cannot fix
would you be able to fix it? just run `make style` and fix the format issues as listed

thanks!

**Comment by TuanNT-ZenAI** [comment]
@yiyixuxu I just fixed

**Comment by xduzhangjiayu** [comment]
> @yiyixuxu I just fixed

Hi,
I have read the code, and i'm confused why `masked_image_latents ` is not used when send to DiT?

**Comment by Trgtuan10** [comment]
@xduzhangjiayu No problem, I don’t think anyone is using these parameters. Feel free to open a PR to make the code clearer

**Comment by xduzhangjiayu** [comment]
@Trgtuan10 We don't need to concat the `masked_image_latents ` and `latents ` for inpainting model?

**Comment by Trgtuan10** [comment]
Yes, we don't need. The mask is only used when creating latents after each step, [code](https://github.com/huggingface/diffusers/blob/828ded2c435cf9a6c67149497d1d1a4d300b3a8a/src/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_inpaint.py#L1054)

**Comment by xduzhangjiayu** [comment]
> Yes, we don't need. The mask is only used when creating latents after each step, [code](https://github.com/huggingface/diffusers/blob/828ded2c435cf9a6c67149497d1d1a4d300b3a8a/src/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_inpaint.py#L1054)

Thanks! I understand, it's like a BLD inpainting

**Comment by huan085128** [comment]
> Yes, we don't need. The mask is only used when creating latents after each step, [code](https://github.com/huggingface/diffusers/blob/828ded2c435cf9a6c67149497d1d1a4d300b3a8a/src/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_inpaint.py#L1054)

I tried to find that there was a problem with edge generation when adding a object, and it also worked when trying multiple random seeds, I have no idea what caused this.

```python
import torch
from diffusers import QwenImageEditInpaintPipeline
from diffusers.utils import load_image
import os

pipe = QwenImageEditInpaintPipeline.from_pretrained("Qwen-Image-Edit", torch_dtype=torch.bfloat16)

pipe.to("cuda:5")
prompt = "添加半个木瓜"
negative_prompt = ""
source = load_image("glasses-1285273_1280 (1).jpg")
mask = load_image("ComfyUI_temp_zqkkx_00002_.png")

image = pipe(
    prompt=prompt,
    negative_prompt=negative_prompt,
    image=source,
    mask_image=mask,
    strength=1.0,
    num_inference_steps=35,
    true_cfg_scale=4.0,
    generator=torch.Generator(device="cuda").manual_seed(41412)
).images[0]
image.save(f"qwen_inpainting_6.png")
```

<p align="center">
  
  
  
</p>

**Comment by Trgtuan10** [comment]
What's wrong with this results @huan085128 ?

**Comment by huan085128** [comment]
> What's wrong with this results @huan085128 ?

Sorry, I didn’t explain it clearly. The issue is at the edges of the generated object—here’s a zoomed-in view：
<p align="center">
  
  
</p>

**Comment by Trgtuan10** [comment]
@huan085128  I’d suggest trying to blur the mask for better results. However, Qwen-Edit is more suited for editing existing content rather than generating entirely new elements.

**Comment by huan085128** [comment]
> @huan085128 I’d suggest trying to blur the mask for better results. However, Qwen-Edit is more suited for editing existing content rather than generating entirely new elements.

Thanks for your reply. I’ve already tried using a blurred mask, but got the same result. Then I tried text editing, and I still noticed some issues around the edges:
prompt: replace the '国' to '打' 
<table>
  
  <tr>
    <td align="center">Source</td>
    <td align="center">Mask</td>
    <td align="center">Qwen edit</td>
  </tr>
  
  <tr>
    <td>
      
    </td>
    <td>
      
    </td>
    <td>
      
    </td>
  </tr>
</table>

**Comment by Trgtuan10** [comment]
May be the inpainting area is too small, model need a bigger space to achive good result. You can think about Adetailer method, using `padding_mask_crop` option
